tests/Unit/OCKL/AsyncStreams/multiple-services.cpp

```diff
@@ -0,0 +1,72 @@
+// RUN: %hc %s -I/opt/rocm/hsa/include -L/opt/rocm/lib -lhsa-runtime64 -o %t.out && %t.out && %t.out -s
+
+#include "as_consumer.h"
+
+class test_handler_t : public ockl_as_consumer_t {
+    bool data[4] = { false, };
+
+    bool check_packet(uint8_t service, uint64_t connection_id,
+                       uint8_t *payload, uint8_t bytes,
+                       uint8_t flags)
+    {
+        if (bytes != 1) return false;
+        if (flags != (__OCKL_AS_CONNECTION_BEGIN | __OCKL_AS_CONNECTION_END))
+            return false;
+        if (service != *payload) return false;
+        if (data[service]) return false;
+        data[service] = true;
+        return true;
+    }
+
+    bool check_final() {
+        if (stream->write_index != 4) return false;
+
+        for (int ii = 0; ii != 4; ++ii) {
+            if (!data[ii]) return false;
+        }
+
+        return true;
+    }
+
+public:
+    test_handler_t(__ockl_as_stream_t *stream) : ockl_as_consumer_t(stream) {};
+};
+
+void multipleServices(__ockl_as_stream_t* stream, unsigned int tid) [[hc]]
+{
+    uint64_t connection_id;
+
+    for (unsigned char ii = 0; ii != 4; ++ii) {
+        __ockl_as_write_block(stream, ii, &connection_id, &ii, 1,
+                              __OCKL_AS_CONNECTION_BEGIN | __OCKL_AS_CONNECTION_END);
+    }
+}
+
+bool run_test()
+{
+    unsigned int num_packets = (synchronous_mode ? 4 : 1);
+    unsigned int num_threads = 1;
```

Comment:
This test is really weak.  Would you make it stronger?

Reply:
Removed this file entirely. All the other files now check for the service ID too.